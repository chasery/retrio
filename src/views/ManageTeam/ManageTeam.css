.ManageTeam__header {
  display: flex;
}
.ManageTeam__header > h2 {
  flex-grow: 1;
}
.ManageTeam__header > a + a {
  margin-left: 0.5rem;
}
