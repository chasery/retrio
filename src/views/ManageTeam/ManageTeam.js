import React, { useEffect, useState } from 'react';
import { Link, useHistory, useParams } from 'react-router-dom';
import TeamsApiService from '../../services/teams-api-service';
import TokenService from '../../services/token-service';
import Header from '../../components/Header/Header';
import ManageTeamList from '../../components/ManageTeamList/ManageTeamList';
import Error from '../../components/Error/Error';
import './ManageTeam.css';

function ManageTeam(props) {
  const history = useHistory();
  const { teamId } = useParams();
  const [team, setTeam] = useState([]);
  const [canModify, setCanModify] = useState(false);
  const [error, setError] = useState(null);

  useEffect(() => {
    async function initState() {
      try {
        let apiCall = await TeamsApiService.getTeam(teamId);
        let res = await apiCall;

        setTeam(res);

        const jwt = TokenService.readJwtToken();
        const user = res.members.find((member) => member.user_id === jwt.id);

        if (user.owner) setCanModify(true);
      } catch (error) {
        setError(error.error);
      }
    }

    initState();
  }, [teamId]);

  const handleDeleteTeam = () => {
    TeamsApiService.deleteTeam(teamId)
      .then((res) => history.push(`/teams`))
      .catch((error) => setError(error.error));
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='ManageTeam'>
          <div className='ManageTeam__wrapper'>
            <div className='ManageTeam__header'>
              <h2>{team ? team.name : 'Manage Team'}</h2>
              {canModify && (
                <>
                  <Link to={`/teams/${teamId}/edit-team`}>Edit Team</Link>
                  <button className='Link' onClick={handleDeleteTeam}>
                    Delete Team
                  </button>
                </>
              )}
            </div>
            {error ? (
              <Error message={error} />
            ) : (
              <>
                <ManageTeamList
                  members={team ? team.members : []}
                  canModify={canModify}
                />
                <Link
                  className='ManageTeam__addTeamMember'
                  to={`/teams/${teamId}/add-team-member`}
                >
                  Add Team Member
                </Link>
              </>
            )}
          </div>
        </section>
      </main>
    </>
  );
}

export default ManageTeam;
