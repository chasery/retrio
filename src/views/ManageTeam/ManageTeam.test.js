import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ManageTeam from './ManageTeam';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    teamId: '0c36af8f-4b3b-4f7ee-9491-c3d6396f1136',
  }),
}));

describe('ManageTeam component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<ManageTeam />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
