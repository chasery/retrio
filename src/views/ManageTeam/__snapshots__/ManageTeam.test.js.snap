// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageTeam component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="ManageTeam"
    >
      <div
        className="ManageTeam__wrapper"
      >
        <div
          className="ManageTeam__header"
        >
          <h2 />
        </div>
        <ManageTeamList
          canModify={false}
        />
        <Link
          className="ManageTeam__addTeamMember"
          to="/teams/0c36af8f-4b3b-4f7ee-9491-c3d6396f1136/add-team-member"
        >
          Add Team Member
        </Link>
      </div>
    </section>
  </main>
</Fragment>
`;
